lib/migrate_user_config.py

```diff
@@ -0,0 +1,156 @@
+"""
+This is called in postinst when upgrading from a version <27.14
+This removes the "ua_config" sub-document from uaclient.conf and moves it to
+it's own json file in /var/lib/ubuntu-advantage/user-config.json.
+The fields under "ua_config" are intended to be set/unset by the user via the
+`pro config set` command, so we don't want them covered as a "conffile".
+It writes back a uaclient.conf that will match the new default uaclient.conf
+as long as no defaults have changed.
+We print warning messages if errors occur with instructions to help the user
+resolve them.
+"""
+
+import json
+import sys
+
+import yaml
+
+from uaclient import defaults, messages
+
+UACLIENT_CONF_BACKUP_PATH = (
+    "/etc/ubuntu-advantage/uaclient.conf.preinst-backup"
```

Comment:
I think we can move this into the defaults module

Reply:
This is only used in this script and only ever will be used in this script, so I'm not sure `defaults`, which is used more broadly, is the right place for it.

Reply:
Good point